cmd/tuple/write.go

```diff
@@ -191,6 +192,22 @@ func writeTuplesFromFile(ctx context.Context, flags *flag.FlagSet, fgaClient *cl
 		outputResponse["failed"] = response.Failed
 	}
 
+	if len(response.Failed) > 0 {
+		failedFormat := formatFromExtension(fileName)
+
+		failedTuples := make([]client.ClientTupleKey, 0, len(response.Failed))
+		for _, f := range response.Failed {
+			failedTuples = append(failedTuples, f.TupleKey)
+		}
+
+		out, errFmt2 := formatTuples(failedTuples, failedFormat)
+		if errFmt2 != nil {
+			return fmt.Errorf("failed to marshal failed tuples: %w", errFmt2)
+		}
+
+		fmt.Fprint(os.Stderr, out)
+	}
```

Comment:
Potential issue in this:

- If formatting fails, the entire command fails, but successful tuples were already written
- No indication to the user that failed tuples are being written to stderr
- Silent failure if stderr write fails


Coudld be something like:

```go
    out, errFmt2 := formatTuples(failedTuples, failedFormat)
    if errFmt2 != nil {
        // Log warning but don't fail the entire operation
        fmt.Fprintf(os.Stderr, "Warning: failed to format failed tuples: %v\n", errFmt2)
    } else {
        if _, err := fmt.Fprint(os.Stderr, out); err != nil {
            fmt.Fprintf(os.Stderr, "Warning: failed to write failed tuples to stderr: %v\n", err)
        }
    }
```

Reply:
FWIW, failed tuples are also written in the standard command output.

```
dist/fga tuple write --file tests/fixtures/basic-tuples.csv --hide-imported-tuples 2> errors.csv 
{
  "failed": [
    {
      "tuple_key": {
        "object":"group:foo",
        "relation":"owner",
        "user":"user:anne"
      },
      "reason":"error message: cannot write a tuple which already exists: user: 'user:anne', relation: 'owner', object: 'group:foo': tuple to be written already existed or the tuple to be deleted did not exist"
    }
  ],
  "failed_count":1,
  "successful_count":0,
  "time_spent":"5.007833ms",
  "total_count":1
}

errors.csv:

user_type,user_id,user_relation,relation,object_type,object_id,condition_name,condition_context
user,anne,,owner,group,foo,,

```

Reply:
Yeah, was asking about the same!